odoo/service/server.py

```diff
@@ -443,9 +443,9 @@ def worker_spawn(self, klass, workers_registry):
 
     def long_polling_spawn(self):
         nargs = stripped_sys_argv()
-        cmd = os.path.join(os.path.dirname(nargs[0]), "odoo.py")
-        nargs = [cmd, 'gevent'] + nargs[1:]
-        popen = subprocess.Popen([sys.executable] + nargs)
+        parent = os.path.dirname(config['root_path'])
+        cmd = [sys.executable, '-m', 'odoo', 'gevent'] + nargs[1:]
```

Comment:
@KangOl `python -m odoo` works in most case. But there is an important development mode scenario that fails.

In more details the failure scenario goes like this:
* create and activate a venv
* from the Odoo source directory, install with `pip install --editable .`
* install some addon package: `pip install ./a1` (from the attached zip: 
[a1a2.zip](https://github.com/odoo/odoo/files/495973/a1a2.zip))

This creates `odoo/addons/a1` in `site-packages`. So at this point you have an `odoo` directory in site-packages but that one does not contain the Odoo source. In such situation `odoo` works (and correctly adds site-packages/odoo/addons to addons-path) but `python -m odoo` fails.

Running Odoo with $VIRTUAL_ENV/bin/odoo works normally because it imports pkg_resources first. So that is fine.

The alternative in https://github.com/odoo-dev/odoo/pull/108 works.